src/elements/Table/TableCell.tsx

```diff
@@ -35,3 +21,7 @@ export function TableCell({ children, node }: Props) {
         </Cell>
     );
 }
+
+export namespace TableCell {
+    export type Node = TableCellNode & { isHeader: boolean };
```

Comment:
By the way, I had a similar challenge in the React emails project, and managed to solve it without extending the model:

https://github.com/prezly/rock-emails/blob/0a96c2d9d10079dc6430fdd07aced3b69c4225fc/lib/renderer/nodes/Table.tsx#L83

Reply:
I wanted to do it the same way you did, by having a single Table component and rendering everything inside, but the problem was the way `Renderer` component is implemented, where it iterates over every node in the document and then renders the component that matches.

I think in order to implement it like you did, we would have to change the whole renderer (or maybe change the `Renderer` to only iterate over top-level nodes) and then have the top-level components render their children separately.